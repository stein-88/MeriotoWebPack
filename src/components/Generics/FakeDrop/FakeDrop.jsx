import React, { useState, useEffect, useRef } from 'react'
import s from './FakeDrop.scss'

const FakeDrop = ({
	selected, items, trocaTexto, dropClass, boxClass, itemClass, sectionClass,
}) => {
	const [mostrardrop, setMostrarDrop] = useState(false)
	const didMount = useRef(true)
	const refEle = useRef()
	const dropRef = useRef(mostrardrop)

	const changeDropRef = (param) => {
		dropRef.current = param
		return setMostrarDrop(param)
	}

	const outClick = (e) => {
		if (!e || !e.target) return null
		if (dropRef.current && refEle.current && !refEle.current.contains(e.target)) {
			return changeDropRef(false)
		}
		return null
	}

	useEffect(() => {
		document.body.addEventListener('mousedown', outClick)
		return () => {
			document.body.removeEventListener('mousedown', outClick)
			didMount.current = false
		}
	}, [])

	if (!items || !items.length || !trocaTexto) return null
	return (
		<section ref={refEle} className={`w-100 text-center ${sectionClass || ''}`}>
			<section
				onClick={() => changeDropRef(!mostrardrop)}
				aria-hidden="true"
				className={boxClass || s.caixinha}
			>
				<p className="mb-0">
					{selected || 'Select one'}
					{!boxClass && <span className="float-right">V</span>}
				</p>
			</section>
			{mostrardrop
				&& (
					<section className={dropClass || s.classedrop}>
						{items.map((ele, i) => (
							<p
								key={`cadaitem${i + 1}`}
								aria-hidden="true"
								onClick={() => {
									trocaTexto(ele)
									if (!mostrardrop) return null
									return changeDropRef(false)
								}}
								className={itemClass || s.item}
							>
								{ele}
							</p>
						))}
					</section>
				)}
		</section>
	)
}

export default FakeDrop
