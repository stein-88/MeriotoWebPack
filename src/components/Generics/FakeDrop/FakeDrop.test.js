import React from 'react'
import renderer from 'react-test-renderer'
import FakeDrop from './FakeDrop'

describe('Test on FakeDrop Component', () => {
    const compFakeDrop = renderer.create(<FakeDrop />)
    test('Should not render without props', () => {
        const jsonFakeDrop = compFakeDrop.toJSON()
        expect(jsonFakeDrop).toBeNull()
    })
    test('Should not render with 1 props only', () => {
        const compFakeDropItems = renderer.create(<FakeDrop items={
            [
                'legume',
                'verdura',
                'fruta'
            ]
        } />)
        const compFakeDropTroca = renderer.create(<FakeDrop trocaTexto={() => null} />)
        const jsonFakeDropItems = compFakeDropItems.toJSON()
        const jsonFakeDropTroca = compFakeDropTroca.toJSON()
        expect(jsonFakeDropItems).toBeNull()
        expect(jsonFakeDropTroca).toBeNull()
    })
    test('Should render with 2 props', () => {
        const compFakeDrop = renderer.create(<FakeDrop items={
            [
                'legume',
                'verdura',
                'fruta'
            ]
        } trocaTexto={() => null} />)
        const jsonFakeDrop = compFakeDrop.toJSON()
        expect(jsonFakeDrop).toMatchSnapshot()
    })
})