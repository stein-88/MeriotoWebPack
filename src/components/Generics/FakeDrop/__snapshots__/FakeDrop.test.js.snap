// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on FakeDrop Component Should render with 2 props 1`] = `
<section
  className="w-100 text-center "
>
  <section
    aria-hidden="true"
    className="caixinha"
    onClick={[Function]}
  >
    <p
      className="mb-0"
    >
      Select one
      <span
        className="float-right"
      >
        V
      </span>
    </p>
  </section>
</section>
`;
